lib/content-negotiation.js

```diff
@@ -0,0 +1,80 @@
+// Copyright IBM Corp. 2016. All Rights Reserved.
+// Node module: strong-error-handler
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+var accepts = require('accepts');
+var debug = require('debug')('strong-error-handler:http-response');
+var sendJson = require('./send-json');
+var sendHtml = require('./send-html');
+
+module.exports = negotiateContentProducer;
+
+/**
+ * Handles req.accepts and req.query._format and options.defaultTypes
+ * to resolve the correct operation
+ *
+ * @param req request object
+ * @param {Object} options options of strong-error-handler
+ * @returns {Function} Opeartion function with signature `fn(res, data)`
+ */
+function negotiateContentProducer(req, options) {
+  var SUPPORTED_TYPES = [
+    'application/json', 'json',
+    'text/html', 'html',
+  ];
+
+  options = options || {};
+  var defaultType = 'json';
+
+  // checking if user provided defaultType is supported
+  if (options.defaultType) {
+    if (SUPPORTED_TYPES.indexOf(options.defaultType) > -1) {
+      debug('Accepting options.defaultType `%s`', options.defaultType);
+      defaultType = options.defaultType;
+    } else {
+      console.warn('defaultType: `%s` is not supported,' +
+        ' falling back to defaultType: `%s`', options.defaultType, defaultType);
+    }
+  }
+
+  // decide to use resolvedType or defaultType
+  // Please note that accepts assumes the order of content-type is provided
+  // in the priority returned
+  // example
+  // Accepts: text/html, */*, application/json ---> will resolve as text/html
+  // Accepts: application/json, */*, text/html ---> will resolve as application/json
+  // Accepts: */*, application/json, text/html ---> will resolve as */*
```

Comment:
@davidcheung in one of your comments, you wrote:

> you're right, glad you asked me to split the double request test cases, the first expect from the request wasn't being looked at by the test case. 
`*/*, text/html` actually returns `text/html` from the **`accepts`** module

I think that means this source-code comment needs to be updated?